CHANGELOG.md

```diff
@@ -84,6 +95,29 @@ them elsewhere before July 2024. Following brownouts in May and June, Teleport
 will disable pulls from all Teleport quay.io repositories on Wednesday July 3,
 2024.
 
+#### Amazon AMIs
+
+Teleport 15 contains several breaking changes to improve the default security
+and usability of Teleport-provided Amazon AMIs.
+
+##### Legacy Amazon Linux 2 AMIs
+
+Teleport-provided Amazon Linux 2 AMIs were deprecated, and Teleport 14 is the
+last version to produce such legacy AMIs. With Teleport 15's release, only
+the newer hardened Amazon Linux 2023 AMIs will be produced.
+
+The legacy AMIs will continue to be published for Teleport 13 and 14 throughout
+the remainder of these releases' lifecycle.
```

Comment:
This change hasn't merged yet afaik. https://github.com/gravitational/teleport/issues/34282

I'd recommend we don't scope creep to include it herein, as there are other docs and example changes needed to complete the work. For instance:

```console
walt@vm:~/git/teleport$ rg 126027368216            
assets/aws/Makefile
44:AWS_ACCOUNT_ID ?= 126027368216

examples/aws/terraform/starter-cluster/README.md
98:# OSS: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss*'
99:# Enterprise: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent*'

examples/aws/terraform/starter-cluster/data.tf
28:  owners      = [126027368216]

examples/aws/terraform/ha-autoscale-cluster/README.md
46:# OSS: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss*'
47:# Enterprise: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent*'

examples/aws/terraform/ha-autoscale-cluster/data.tf
6:  owners      = [126027368216]

docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx
169:The AWS account ID which publishes these AMIs is `126027368216`. You can list the available AMIs with
178:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss-(=teleport.version=)'
183:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)'
188:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)-fips'

docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx
182:The AWS account ID which publishes these AMIs is `126027368216`. You can list the available AMIs with
191:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss-(=teleport.version=)'
196:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)'
201:  aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)-fips'
```

We can pick up the legacy AMI deprecation in the change that removes/fixes the above.